File: src/frontend/src/app/components/atendimento/agendamento/agendamento/model/agendamento-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendamentoGridDTO {
  @Expose() id?: string;
  @Expose() agendaNome?: string;
  @Expose() pacienteNome?: string;
  @Expose() convenioNome?: string;
  @Expose() procedimentoNome?: string;
  @Expose() status?: string;
  @Expose() primeiraData?: string;
  @Expose() qtdHorarios?: number;
  @Expose() deleted?: boolean;
  @Expose() createdAt?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agendamento/dto/AgendamentoGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agendamento.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendamentoGridDTO implements GridDTO {

    private UUID id;
    private String agendaNome;
    private String profissionalNome;
    private String pacienteNome;
    private String convenioNome;
    private String procedimentoNome;
    private String status;
    private LocalDate primeiraData;
    private LocalDateTime primeiraDataHora;
    private int qtdHorarios;
    private Boolean deleted;
    private LocalDateTime createdAt;
}
